### PR TITLE
docs: add Web UI requirements spec and update design/validation trifecta

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Summary

Creates a clean requirements/design/validation trifecta for the Web UI SPA.

### New files
- **\docs/web-ui-requirements.md\** ΓÇö 53 formal requirements (WEB-0101 through WEB-CC-04) extracted from the existing design doc, \deploy/web-ui/app.js\, and validation plan. Every requirement has: ID, description with RFC 2119 keywords, acceptance criteria, priority, source citation, and confidence level.
- **\docs/audits/web-ui-extraction-audit.md\** ΓÇö Extraction audit report. 11 findings, all remediated. Verdict: PASS.

### Updated files
- **\docs/web-ui-design.md\** ΓÇö Fixed 3 inconsistencies found during audit:
  - \sessionStorage\ clearing scope: now specifies MSAL-related keys only (matches code)
  - Redirect URI: now describes path normalization (stripping filename component)
  - Removed stale \programroute\ table reference from architecture diagram
  - Updated cross-references to new requirements doc
- **\docs/web-ui-validation.md\** ΓÇö Restructured with proper sections (Overview, Scope, Test Strategy, Traceability Matrix, Risk-Based Prioritization, Pass/Fail Criteria). Added 13 new test cases covering:
  - Schedule divergence edge cases (T-WEB-0107, T-WEB-0108)
  - Graph limits: 20-series max, 500-point downsampling, tooltip (T-WEB-0713ΓÇô0715)
  - Environment validation: GUID format, storage account, function app name (T-WEB-0807ΓÇô0809)
  - Redirect URI filename stripping (T-WEB-0509)
  - Cross-cutting concerns: zero-build, XSS escaping, MSAL hash routing, popup detection (T-WEB-CC-01ΓÇô04)

### Requirement coverage
| Section | Count |
|---------|-------|
| Dashboard (WEB-0100) | 4 |
| Desired State (WEB-0200) | 7 |
| Program Ingest (WEB-0300) | 10 |
| Program List (WEB-0400) | 1 |
| Authentication (WEB-0500) | 8 |
| Infrastructure (WEB-0600) | 6 |
| Sensor Data (WEB-0700) | 4 |
| Environment Manager (WEB-0800) | 7 |
| Deployment (WEB-0900) | 2 |
| Cross-Cutting | 4 |
| **Total** | **53** |

### Methodology
Spec extraction workflow: repository scan ΓåÆ draft extraction ΓåÆ consistency audit ΓåÆ remediation. All requirements trace to specific code or design doc text (High confidence).

### CODE REVIEW

File: docs/audits/web-ui-extraction-audit.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+---
+
+## 1  Executive Summary
+
+The Web UI spec trifecta is internally consistent after remediation. The
+extraction produced 42 formal requirements (WEB-0101 through WEB-CC-04),
+all at High confidence, traced to code and design documentation. Three
+blocking inconsistencies were found and fixed during the audit; 8
+non-blocking gaps were addressed by adding 13 new test cases. The spec
+set is ready for approval.
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/audits/web-ui-extraction-audit.md]: Fixed — the audit report now correctly states 53 formal requirements. The count was a stale reference from an earlier draft.

File: docs/web-ui-requirements.md [hunk 2]
```diff
@@ -0,0 +1,1112 @@
...
+**Confidence:** High
+
+**Description:**
+The SPA MUST provide a Sensor Data tab that loads data from the `SensorData`
+Azure Table.
+
+**Acceptance criteria:**
+
+1. A "Sensor Data" tab appears in the tab bar.
+2. The tab queries `SensorData` table using per-node partition queries.
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-requirements.md]: Fixed — aligned all references to \sensordata\ (lowercase) matching the deployed config in \pp.js\ and Bicep defaults. Updated in requirements, design, and validation docs.

File: docs/web-ui-design.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ Browser (SPA)
 ├── Dashboard (read actualstate table)
 ├── Desired State (read/write desiredstate table)
 ├── Program Upload (POST ELF to ProgramIngest function)
-├── Program List (read programs + programroute tables)
+├── Program List (read programs table)
 └── Sensor Data (read SensorData table, time-series graph)
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-design.md]: Fixed — architecture diagram now uses \sensordata\ (lowercase) matching the SPA config and Bicep provisioning.

File: docs/web-ui-validation.md [hunk 4]
```diff
@@ -2,60 +2,191 @@
...
+| WEB-0603 | T-WEB-0603 |
+| WEB-0604 | T-WEB-0604 |
+| WEB-0605 | T-WEB-0605 |
+| WEB-0606 | T-WEB-0606 |
+| WEB-0607 | T-WEB-0607 |
+| WEB-0700 | T-WEB-0701 |
+| WEB-0701 | T-WEB-0702, T-WEB-0703, T-WEB-0706, T-WEB-0713, T-WEB-0714, T-WEB-0715 |
+| WEB-0702 | T-WEB-0704, T-WEB-0705 |
+| WEB-0703 | T-WEB-0707, T-WEB-0708, T-WEB-0709, T-WEB-0710, T-WEB-0711, T-WEB-0712 |
+| WEB-0800 | — (covered by WEB-0801 through WEB-0806 tests) |
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-validation.md]: Fixed — mapped WEB-0800 to T-WEB-0801 through T-WEB-0806 in the traceability matrix instead of the dash placeholder.

File: docs/web-ui-validation.md [hunk 5]
```diff
@@ -2,60 +2,191 @@
...
+- SPA front-end behavior (all tabs)
+- Environment manager
+- Authentication flows (MSAL.js)
+- Program upload flow (SPA → ProgramIngest)
+- Infrastructure provisioning (Bicep, CORS, EasyAuth, GitHub Pages)
+- Cross-cutting concerns (escaping, MSAL hash routing, popup detection)
+
+### Out of scope
+
+- Azure handler function internals (covered by azure-handler-validation.md)
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-validation.md]: Fixed — clarified to 'Azure handler function internals beyond ProgramIngest' so the scope is consistent with including T-WEB-0301–0308.

File: docs/web-ui-requirements.md [hunk 6]
```diff
@@ -0,0 +1,1112 @@
...
+- Authentication via MSAL.js (Entra ID)
+- Program upload flow (SPA → ProgramIngest Azure Function)
+- Infrastructure requirements (Bicep provisioning, CORS, EasyAuth, GitHub Pages deployment)
+
+### Out of scope
+
+- Azure handler function internals (covered by [azure-handler-requirements.md](azure-handler-requirements.md))
+- Gateway, node, or modem firmware behavior
+- BPF program compilation or Prevail verifier internals
+- Azure Table schema design (owned by the handler)
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-requirements.md]: Fixed — reworded to 'Azure handler function internals beyond ProgramIngest''s externally observable contracts' to avoid contradicting the WEB-0300 section.

File: docs/web-ui-requirements.md [hunk 7]
```diff
@@ -0,0 +1,1112 @@
...
+
+**Description:**
+The SPA MUST provide a Sensor Data tab that loads data from the `sensordata`
+Azure Table.
+
+**Acceptance criteria:**
+
+1. A "Sensor Data" tab appears in the tab bar.
+2. The tab queries `sensordata` table using per-node partition queries.
+3. The tab displays a loading indicator while fetching.
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-requirements.md:737]: Added a note at WEB-0700 explaining the lowercase convention and that Azure Table names are case-insensitive, so readers won't be confused by other docs using the capitalized form.

File: docs/web-ui-requirements.md [hunk 8]
```diff
@@ -0,0 +1,1117 @@
...
+
+## 4  Requirement Format
+
+Each requirement uses the following fields:
+
+- **ID** — Unique identifier (`WEB-XXYY`).
+- **Title** — Short name.
+- **Description** — What the SPA must do, using RFC 2119 keywords.
+- **Acceptance criteria** — Observable, testable conditions (AC-1, AC-2, …).
+- **Priority** — MoSCoW: **Must**, **Should**, **May**.
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-requirements.md:72]: Fixed — updated the format description to document both \WEB-XXYY\ (functional) and \WEB-CC-NN\ (cross-cutting) ID patterns.

File: docs/web-ui-requirements.md [hunk 9]
```diff
@@ -0,0 +1,1117 @@
...
+**Source:** web-ui-design.md §9.3
+**Confidence:** High
+
+**Description:**
+The ProgramIngest function's `authLevel` MUST be `anonymous` — authentication
+is delegated to EasyAuth, not Azure Functions runtime keys.
+
+**Acceptance criteria:**
+
+1. `ProgramIngest/function.json` specifies `authLevel: "anonymous"`.
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-requirements.md]: Fixed — updated to the full repo path: \crates/sonde-azure-handler/function-app/ProgramIngest/function.json\.

File: docs/web-ui-design.md [hunk 10]
```diff
@@ -158,8 +158,9 @@ local `ProgramLibrary`.
 - MSAL.js 2.x with authorization code flow + PKCE.
 - Token caching in browser session storage.
 - Silent token renewal; redirect to login on expiry.
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/web-ui-design.md]: Fixed — updated design doc §8 to describe the actual popup fallback behavior: \cquireTokenSilent\ with \cquireTokenPopup\ fallback. No redirects are used.

File: docs/audits/web-ui-extraction-audit.md [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+---
+
+## 1  Executive Summary
+
+The Web UI spec trifecta is internally consistent after remediation. The
+extraction produced 53 formal requirements (WEB-0101 through WEB-CC-04),
+all at High confidence, traced to code and design documentation. Three
+blocking inconsistencies were found and fixed during the audit; 8
+non-blocking gaps were addressed by adding 13 new test cases. The spec
+set is ready for approval.
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply docs/audits/web-ui-extraction-audit.md:18]: Fixed — updated the PR description from 42 to 53 requirements.
